src/entities/trade.ts

```diff
@@ -232,19 +256,42 @@ export class Trade<TInput extends Currency, TOutput extends Currency, TTradeType
    * @param tradeType The type of trade, exact input or exact output
    */
   private constructor({
-    route,
+    routes,
     inputAmount,
     outputAmount,
     tradeType
   }: {
-    route: Route<TInput, TOutput>
+    routes: { percent: Percent; route: Route<TInput, TOutput> }[]
     inputAmount: CurrencyAmount<TInput>
     outputAmount: CurrencyAmount<TOutput>
     tradeType: TTradeType
   }) {
-    invariant(inputAmount.currency.equals(route.input), 'INPUT_CURRENCY_MATCH')
-    invariant(outputAmount.currency.equals(route.output), 'OUTPUT_CURRENCY_MATCH')
-    this.route = route
+    invariant(
+      routes.every(({ route }) => inputAmount.currency.equals(route.input)),
+      'INPUT_CURRENCY_MATCH'
+    )
+    invariant(
+      routes.every(({ route }) => outputAmount.currency.equals(route.output)),
+      'OUTPUT_CURRENCY_MATCH'
+    )
+
+    const totalPercent = routes
+      .map(({ percent }) => percent)
+      .reduce((total, cur) => total.add(cur), new Percent(0, 100))
+
+    invariant(totalPercent.equalTo(new Percent(100, 100)), 'TOTAL_PERCENT')
```

Comment:
hmm, i guess we could ensure full precision here. for e.g. exact input trades, you could just ensure that for each percent in routes, there's no remainder from multiplying.

Reply:
I added various checks of this nature to the fromRoutes method and the constructor itself